File: server/src/models/album_model.py
from sqlalchemy.orm import relationship
from sqlalchemy import Column, String, Integer, DateTime, ForeignKey, func

from src.database import Base

class Album(Base):
  __tablename__ = 'album'
  album_id = Column(Integer, primary_key=True, index=True, autoincrement=True)
  user_id = Column(Integer, ForeignKey('user.user_id', ondelete='CASCADE'), nullable=True) 
  album_cover = Column(String(1000), index=True)
  album_name = Column(String(50), nullable=False, index=True)
  description = Column(String(1000), nullable=False, index=True)
  created_at = Column(DateTime(timezone=True), default=func.now())
  modified_at = Column(DateTime(timezone=True), default=func.now(), onupdate=func.now())

  user = relationship('User', back_populates='album')
  audio = relationship('Audio', back_populates='album', uselist=False)

File: audioloca/lib/local/models/album.model.dart
class Album {
  final int albumId;
  final String albumName;
  final String? albumCover;
  final DateTime createdAt;
  final DateTime modifiedAt;

  Album({
    required this.albumId,
    required this.albumName,
    required this.albumCover,
    required this.createdAt,
    required this.modifiedAt,
  });

  factory Album.fromJson(Map<String, dynamic> json) {
    return Album(
      albumId: json['album_id'] ?? 0,
      albumName: json['album_name'] ?? '',
      albumCover: json['album_cover'] ?? '',
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : DateTime.fromMillisecondsSinceEpoch(0),
      modifiedAt: json['modified_at'] != null
          ? DateTime.parse(json['modified_at'])
          : DateTime.fromMillisecondsSinceEpoch(0),
    );
  }
  static Album empty() {
    return Album(
      albumId: 0,
      albumName: '',
      albumCover: null,
      createdAt: DateTime.fromMillisecondsSinceEpoch(0),
      modifiedAt: DateTime.fromMillisecondsSinceEpoch(0),
    );
  }
}
